src/components/AddItem.jsx

```diff
@@ -1,7 +1,102 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { collection, addDoc } from 'firebase/firestore';
+import { db } from '../lib/firebase';
+
+const radioButtonOptions = [
+  {
+    option: 'soon',
+    value: 7,
+    defaultChecked: true,
+  },
+  {
+    option: 'kind of soon',
+    value: 14,
+    defaultChecked: false,
+  },
+  {
+    option: 'not soon',
+    value: 30,
+    defaultChecked: false,
+  },
+];
 
 function AddItem() {
-  return <div>Add Item</div>;
+  const [itemName, setItemName] = useState('');
+  const [frequency, setFrequency] = useState(7);
+  const [notification, setNotification] = useState('');
+
+  //retrive the token from localStorage
+  const token = localStorage.getItem('list-token');
+
+  //handle submit for the form
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    //create new doc reference for text input and save to firestore database
+    //use a token to save new items into specific collection (shopping list)
+    try {
+      const docRef = await addDoc(collection(db, token), {
+        //data points being sent to firebase, object format
+        //should we convert frequency to a number? or send in as a string?
+        itemName: itemName,
+        frequency: Number(frequency),
+        purchasedDate: null,
+      });
+      setNotification(`Successfully added ${itemName}`);
+      setItemName('');
+      console.log('Document written with ID: ', docRef.id);
+      //setTimeout is used to clear the success notification after 2 seconds
```

Comment:
This is a great use of comments so someone doesn't have to think too much about the 2000 listed below. Another great alternative is naming the variable so: const twoSecondDelay = 2000; and then using twoSecondDelay in the setTimeout parameter.